core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -94,9 +98,23 @@ private HTTPClient(
 
     HttpClientBuilder clientBuilder = HttpClients.custom();
 
+    String connectionTimeout = properties.get(REST_CONNECTION_TIMEOUT);
+    String socketTimeout = properties.get(REST_SOCKET_TIMEOUT);
+
+    ConnectionConfig.Builder connConfigBuilder = ConnectionConfig.custom();
+
+    if (connectionTimeout != null) {
+      connConfigBuilder.setConnectTimeout(Long.parseLong(connectionTimeout), TimeUnit.SECONDS);
```

Comment:
Seems that iceberg sdk has its own way to parse values from the properties, like the one below:
```
PropertyUtil.propertyAsInt(properties, REST_MAX_RETRIES, 5)
```

Reply:
Yeah, I saw this as well.  I chose not to use this to avoid setting mandatory defaults. We will fallback to actual httpclient defaults this way incase these configs are not set, which is the existing behavior.

Reply:
Let's use PropertyUtils anyways for consistency. We can use PropertyUtil.propertyAsNullableInt to avoid specifying a default

Reply:
Or we can just use the default value of the httpclient?

Reply:
> Or we can just pass the default value of the httpclient to this function?

@XJDKC If there's a way to get this value without hardcoding it then this should work. I'll look into this

> We can use PropertyUtil.propertyAsNullableInt to avoid specifying a default. 

@sfc-gh-mparmar This returns null if the value is not present in the map :/

Reply:
@sfc-gh-mparmar @XJDKC 

It is really difficult to test the code if the logic is in the constructor.
I've refactored the code and moved the ConnectionConfig creation to a separate testable method.

Reply:
I see, the default value is a private variable in ConnectionConfig.Builder. Maybe we can copy that value to iceberg, I think it's clearer if we store that value in iceberg and set it explicitly. Any thought on this?

Reply:
Looks like connection timeout has a default value of 3 minutes and I don't think there's any default for the socket timeout as such. 

Is there any benefit of changing the current behavior by explicitly setting default values by copying them over? 
Not sure if I fully get the benefit of doing this as opposed to falling back to current behavior in case we don't pass these values explicitly. 

Maybe @sfc-gh-mparmar or @dennishuo can shed some light on this? I'm happy to define explicit defaults just curious to know why :)

Reply:
Let's NOT keep a separate copy of default values in iceberg but rather override default only when timeout is specified.